src/cloud.jl

```diff
@@ -63,9 +63,15 @@ PointCloud{T <: AbstractFloat}(points::Matrix{T}) = PointCloud(convert_positions
 
 Convert a standard julia array of points into a FixedSizeArray
 """
-function convert_positions{T <: AbstractFloat}(points::Matrix{T})
+function convert_positions{T}(points::Matrix{T})
     ndims = size(points, 1)
-    return [SVector{ndims,T}(points[:, i]) for i = 1:size(points, 2)]
+    npoints = size(points, 2)
+    if isbits(T)
+        data = reinterpret(SVector{ndims, T}, points, (npoints, ))
+    else
+        data = SVector{ndims, T}[SVector{ndims, T}(points[:, i]) for i in 1:npoints]
```

Comment:
If you care about speed here, it's faster to be explicit - unwrap the `:` and use inbounds (challenging here with the comprehension - let's experiment tomorrow)

Also, did they untyped comprehension have inference errors, out of curiousity?